### PR TITLE
Use case clarification comments #36785

### PR DESCRIPTION
**Pull request by abhilash1910**
Changes in comments for learning rate =1.0 and epsilon =1e-6 following PR #36785. Added description to suit the use cases where the modifications are required as mentioned in the sections if the paper by M. D. Zeiler

### CODE REVIEW

File: tensorflow/python/keras/optimizer_v2/adadelta.py [hunk 1]
```diff
@@ -51,7 +51,37 @@ class Adadelta(optimizer_v2.OptimizerV2):
...
-
+  
+  The learning rate and epsilon according to the original paper uses learning rate as 1.0 and epsilon as 1e-6.
+  
+  According to section 4.3 ("Effective Learning rates"), near the end of 
+  training step sizes converge to 1 which is effectively a high learning 
+  rate which would cause divergence.This occurs only near the end of the
+  training as gradients and step sizes are small,and the epsilon constant 
+  in the numerator and denominator dominate past gradients and parameter
+  Updates which converge learning rate to 1.This can be further
```
Comment (tanzhenyu) [inline tensorflow/python/keras/optimizer_v2/adadelta.py]: updates

Comment (tanzhenyu) [inline tensorflow/python/keras/optimizer_v2/adadelta.py]: space here

File: tensorflow/python/keras/optimizer_v2/adadelta.py [hunk 2]
```diff
@@ -51,7 +51,37 @@ class Adadelta(optimizer_v2.OptimizerV2):
...
+  
+  The learning rate and epsilon according to the original paper uses learning rate as 1.0 and epsilon as 1e-6.
+  
+  According to section 4.3 ("Effective Learning rates"), near the end of 
+  training step sizes converge to 1 which is effectively a high learning 
+  rate which would cause divergence.This occurs only near the end of the
+  training as gradients and step sizes are small,and the epsilon constant 
+  in the numerator and denominator dominate past gradients and parameter
+  Updates which converge learning rate to 1.This can be further 
+  ($$\Delta x_t = -RMS[\Delta x]_{t-1} * g_t / RMS[g]_t$$)
```
Comment (tanzhenyu) [inline tensorflow/python/keras/optimizer_v2/adadelta.py]: this can be removed.

Comment (abhilash1910) [reply tensorflow/python/keras/optimizer_v2/adadelta.py]: Yes ,corrected@tanzhenyu

File: tensorflow/python/keras/optimizer_v2/adadelta.py (hunk 1 above)
Comment (abhilash1910) [reply tensorflow/python/keras/optimizer_v2/adadelta.py]: Yes @tanzhenyu,it is changed

**Comment by abhilash1910** [comment]
Hi @tanzhenyu ,changes are made. Can you review it ? Thanks

**Review by tanzhenyu — DISMISSED** [review]

**Comment by gbaned** [comment]
@abhilash1910 Can you please address Ubuntu Sanity errors? Thanks!

**Comment by abhilash1910** [comment]
Hi @gbaned, according to the summary results for sanity check ,there is FAIl mentioned for do_pylint: Python 3 pylint(1 failed and 14 passed) . I am not sure how to resolve this as it is pretty much the same for all the other accepted PRs.Since the additions are only in form of comments. Can you suggest ways to resolve this? Thanks

**Comment by gbaned** [comment]
@tanzhenyu Can you please assist with build failures. Thanks!

**Comment by tanzhenyu** [comment]
> Hi @gbaned, according to the summary results for sanity check ,there is FAIl mentioned for do_pylint: Python 3 pylint(1 failed and 14 passed) . I am not sure how to resolve this as it is pretty much the same for all the other accepted PRs.Since the additions are only in form of comments. Can you suggest ways to resolve this? Thanks

It seems a temporary failure due to connection errors. Can you re-try?

**Comment by abhilash1910** [comment]
Hi @tanzhenyu ,reuploaded the changes.

File: tensorflow/python/keras/optimizer_v2/adadelta.py [hunk 3]
```diff
@@ -51,7 +51,33 @@ class Adadelta(optimizer_v2.OptimizerV2):
...
+  SGD, ADAGRAD, Momentum.
+  
+  According to section 4.4("Speech Data"),where a large neural network with 
+  4 hidden layers was trained on a corpus of US English data,ADADELTA was 
+  used with 100 network replicas.The epsilon used is 1e-6 with rho=0.95 
+  which converged faster than ADAGRAD.
+  
+  The alternate function declaration can be used for testing the use case 
+  similar to the in the paper
+  def __init__(self, lr=1.0, rho=0.95, epsilon=1e-6, decay=0., **kwargs):
```
Comment (tanzhenyu) [inline tensorflow/python/keras/optimizer_v2/adadelta.py]: please wrap this in:
```
def __init__(xxx)
```

**Review by tanzhenyu — DISMISSED** [review]

File: tensorflow/python/keras/optimizer_v2/adadelta.py (hunk 3 above)
Comment (abhilash1910) [reply tensorflow/python/keras/optimizer_v2/adadelta.py]: Wrapped up in __init__() method

File: tensorflow/python/keras/optimizer_v2/adadelta.py [hunk 4]
```diff
@@ -51,7 +51,7 @@ class Adadelta(optimizer_v2.OptimizerV2):
   $$\Delta x_t = -RMS[\Delta x]_{t-1} * g_t / RMS[g]_t$$
   $$E[\Delta x^2]_t := \rho * E[\Delta x^2]_{t-1} + (1 - \rho) * \Delta x_t^2$$
   $$x_t := x_{t-1} + \Delta x_{t}$$
-
+
```
Comment (tanzhenyu) [inline tensorflow/python/keras/optimizer_v2/adadelta.py]: wrong ident here.

**Review by tanzhenyu — DISMISSED** [review]

File: tensorflow/python/keras/optimizer_v2/adadelta.py (hunk 4 above)
Comment (abhilash1910) [reply tensorflow/python/keras/optimizer_v2/adadelta.py]: Indentation Resolved

**Comment by gbaned** [comment]
@abhilash1910 Can you please resolve conflicts? Thanks!

**Comment by abhilash1910** [comment]
Resolved @gbaned .Thanks!

**Comment by gbaned** [comment]
@abhilash1910 Can you please address Ubuntu Sanity errors? Thanks!

**Comment by abhilash1910** [comment]
> > Hi @gbaned, according to the summary results for sanity check ,there is FAIl mentioned for do_pylint: Python 3 pylint(1 failed and 14 passed) . I am not sure how to resolve this as it is pretty much the same for all the other accepted PRs.Since the additions are only in form of comments. Can you suggest ways to resolve this? Thanks
> 
> It seems a temporary failure due to connection errors. Can you re-try?

I can try re-uploading since CI build logs indicate failure on PyLint 3. It is the same issue which was happening previously.

**Comment by gbaned** [comment]
@abhilash1910 Can you please resolve conflicts? Thanks!

**Comment by abhilash1910** [comment]
> @abhilash1910 Can you please resolve conflicts? Thanks!
Hi @gbaned ,merge conflict resolved.

File: tensorflow/python/keras/optimizer_v2/adadelta.py [hunk 5]
```diff
@@ -45,22 +45,18 @@ class Adadelta(optimizer_v2.OptimizerV2):
...
-    epsilon: A `Tensor` or a floating point value.  A constant epsilon used
-             to better conditioning the grad update.
-    name: Optional name prefix for the operations created when applying
-      gradients.  Defaults to `"Adadelta"`.
-    **kwargs: Keyword arguments. Allowed to be one of
-      `"clipnorm"` or `"clipvalue"`.
-      `"clipnorm"` (float) clips gradients by norm; `"clipvalue"` (float) clips
-      gradients by value.
-
-  Reference:
```
Comment (tanzhenyu) [inline tensorflow/python/keras/optimizer_v2/adadelta.py:62]: I think you should revert this, this resolves the merge conflict by going to the previous version, instead of latest version

Comment (abhilash1910) [reply tensorflow/python/keras/optimizer_v2/adadelta.py:62]: Reverted.Thanks!

**Comment by rthadur** [comment]
Seems auto-merge is not happening but the changes are now committed so we can close this. Thank you for the PR.

**Comment by tanzhenyu** [comment]
Yep I resolved all conflicts internally and mark it contributed by @abhilash1910

**Comment by abhilash1910** [comment]
Thank you @tanzhenyu @gbaned @rthadur.
